src/CalcManager/Ratpack/ratpak.h

```diff
@@ -56,20 +58,15 @@ typedef enum eANGLE_TYPE ANGLE_TYPE;
 //
 //-----------------------------------------------------------------------------
 
-#pragma warning(push)
-#pragma warning(disable : 4200) // nonstandard extension used : zero-sized array in struct/union
 typedef struct _number
 {
-    int32_t sign;   // The sign of the mantissa, +1, or -1
-    int32_t cdigit; // The number of digits, or what passes for digits in the
-                    // radix being used.
-    int32_t exp;    // The offset of digits from the radix point
-                    // (decimal point in radix 10)
-    MANTTYPE mant[];
-    // This is actually allocated as a continuation of the
-    // NUMBER structure.
-} NUMBER, *PNUMBER, **PPNUMBER;
-#pragma warning(pop)
+    int32_t sign = 1;   // The sign of the mantissa, +1, or -1
+    int32_t cdigit = 0; // The number of digits, or what passes for digits in the
+                        // radix being used.
+    int32_t exp = 0;    // The offset of digits from the radix point
+                        // (decimal point in radix 10)
+    std::vector<MANTTYPE> mant;
```

Comment:
It seems `cdigit` is used as count of entries in the flexible array in the long form, is that correct? Additionally, `cdigits` usage seems to suggest such arrays used to have at least one element defaulting to zero, is that correct? Since `vector` by default doesn't contain any values, perhaps default-initialising it with a single zero would address the current problems:
```suggestion
    std::vector<MANTTYPE> mant{0};
```